authz/providers/azure/rbac/rbac.go

```diff
@@ -321,31 +364,33 @@ func (a *AccessInfo) sendCheckAccessRequest(checkAccessURL url.URL, checkAccessB
 		klog.V(10).Infof("binary data:%s", binaryData)
 	}
 
-	req, err := http.NewRequest(http.MethodPost, checkAccessURL.String(), buf)
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, checkAccessURL.String(), buf)
 	if err != nil {
-		reviewResult.err = errors.Wrap(err, "error creating check access request")
-		ch <- reviewResult
-		return
+		return errutils.WithCode(errors.Wrap(err, "error creating check access request"), http.StatusInternalServerError)
 	}
 
 	a.setReqHeaders(req)
-
+	internalServerCode := azureutils.ConvertIntToString(http.StatusInternalServerError)
+	// start time to calculate checkaccess duration
+	start := time.Now()
 	resp, err := a.client.Do(req)
+	duration := time.Since(start).Seconds()
 	if err != nil {
-		reviewResult.err = errors.Wrap(err, "error in check access request execution")
-		ch <- reviewResult
-		return
+		checkAccessTotal.WithLabelValues(internalServerCode).Inc()
+		checkAccessDuration.WithLabelValues(internalServerCode).Observe(duration)
+		return errutils.WithCode(errors.Wrap(err, "error in check access request execution"), http.StatusInternalServerError)
```

Comment:
Will gate way error & client time out be part of this error? If so, should we try to retrieve error code?

Reply:
resp and error cannot be non-nil at the same time